packages/next-auth/src/providers/hubspot.ts

```diff
@@ -0,0 +1,104 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+interface HubSpotProfile extends Record<string, any> {
+
+  // TODO: figure out additional fields, for now using 
+  // https://legacydocs.hubspot.com/docs/methods/oauth2/get-access-token-information
+
+  user: string,
+  user_id: string,
+
+  hub_domain: string,
+  hub_id: string,
+}
+
+
+const HubSpotConfig = {
+  clientId: process.env.HUBSPOT_CLIENT_ID ?? "DEFAULT_CLIENT_ID",
+  clientSecret: process.env.HUBSPOT_CLIENT_SECRET ?? "DEFAULT_CLIENT_SECRET",
+  redirectUri: process.env.HUBSPOT_REDIRECT_URI ?? "DEFAULT_REDIRECT_URI",
+  authorizationUrl: "https://app.hubspot.com/oauth/authorize",
+  tokenUrl: "https://api.hubapi.com/oauth/v1/token",
+  profileUrl: "https://api.hubapi.com/oauth/v1/access-tokens/"
```

Comment:
Shouldn't it be [this endpoint](https://legacydocs.hubspot.com/docs/methods/get-account-details)?

Reply:
@ThangHuuVu the `/me` endpoint provides even less info as compared to `/access-tokens` and it does not contain the `user_id` which is useful

Reply:
So I just revisited their updated docs and see that [this endpoint](https://developers.hubspot.com/docs/api/settings/user-provisioning) contains the user ID, I suggest you use it instead of the access-tokens one

Reply:
@ThangHuuVu not sure what is the benefits of using it. It still does not provide additional info related to the user.

Reply:
@ThangHuuVu I can keep an eye on the update they have in the future and update this as soon as they have a better endpoint for the user info. Would that work to have this resolved?